### PR TITLE
Correcting online tutorial (#1646)

### LINKED ISSUE #1646: The Guide is out of date with respect to auto-ghost
In [Exercise 5 of the Online Guide](https://github.com/dafny-lang/dafny/blob/c8e5320b6454bee20ea56d3ac3bb35f681727a0c/docs/OnlineTutorial/guide.md?plain=1#L804-L808), Dafny no longer reports an error because the variable is instead automatically declared ghost. Instead, users will only get an error if they later try to use the variable in a non-ghost context.

Reported on [Stack Overflow](https://stackoverflow.com/questions/70344332/functions-to-appear-outside-of-annotations)

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #1646

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/OnlineTutorial/guide.md [hunk 1]
```diff
@@ -798,7 +798,7 @@ reference for details).
 
 **Exercise 5.**
   *Change your test method from Exercise 4 to capture the value of
-  `max` to a variable, and then do the checks from Exercise 4 using
+  `max` to an out parameter, and then do the checks from Exercise 4 using
```
Comment (MikaelMayer) [inline docs/OnlineTutorial/guide.md:801]:
```suggestion
  `max` to an out-parameter, and then do the checks from Exercise 4 using
```

File: docs/OnlineTutorial/guide.md [hunk 2]
```diff
@@ -807,8 +807,8 @@ function max(a: int, b: int): int
 {
   // Use your code from Exercise 4
 }
-method Testing() {
-  // Add assertions to check max here. Be sure to capture it to a local variable
+method Testing() returns (r: int) {
+  // Add assertions to check max here. Be sure to capture it to the out parameter
```
Comment (MikaelMayer) [inline docs/OnlineTutorial/guide.md:811]:
```suggestion
  // Add assertions to check max here. Be sure to capture it to the out-parameter
```
Also, there is one more occurrence of "out parameter" in the doc in section 13.3, can you also fix it?
There are 53 occurrences of "out-parameter" (with the dash) in the documentation, so I assume it's the correct one.

**Review by MikaelMayer — COMMENTED** [review]
Just one tweak before I approve it ! It's clever to make a variable an out-parameter, that way you can control the ghostness

File: docs/OnlineTutorial/guide.md (hunk 2 above)
Comment (davidcok) [reply docs/OnlineTutorial/guide.md]: Fixed the other occurrence as well.

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**